File: Core/TouchToolkit/CS/TouchToolkitDemo/GcZoomPages/ControlBarDemo.cs
using System;
using System.Windows.Forms;
using TouchToolkitDemo.C1ZoomPages.Help_Forms;
using System.Threading.Tasks;

namespace TouchToolkitDemo.C1ZoomPages
{
    public partial class ControlBarDemo : DemoBase
    {
        public ControlBarDemo()
        {
            InitializeComponent();

            this.Title = "ControlBar";
            this.Description = @"ControlBar can provide a floating, translucent window for form.
- ControlBar provides built-in actions, the map demo shows how to use the built-in actions.
- Of course, customize the action is simple, keyboard demo shows how to costomize the actions.";
        }

        private async void button1_Click(object sender, EventArgs e)
        {
            var splash = new SplashForm();
            var splashTask = ShowDialogAsync(splash);

            var form = await CreateMapFormAsync();

            splash.Close();
            splash.Dispose();
            await splashTask;

            form.Show(this);
        }

        private void button3_Click(object sender, EventArgs e)
        {
            Form form = new InputFormforControlBar();
            form.Show();
        }

        private async Task<Form> CreateMapFormAsync()
        {
            Form form = null;
            await Task.Run(() => form = new MapViewFormForControlBar());
            return form;
        }

        private async Task<DialogResult> ShowDialogAsync(Form form)
        {
            await Task.Yield();
            if (form.IsDisposed)
                return DialogResult.OK;
            return form.ShowDialog();
        }
    }
}

File: NetFramework/TouchToolkit/VB/TouchToolkitDemo/GcZoomPages/ControlBarDemo.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports C1.Win.TouchToolKit

Imports TouchToolKitDemoVB.C1ZoomPages.Help_Forms

Namespace C1ZoomPages
    Partial Public Class ControlBarDemo
        Inherits DemoBase
        Public Sub New()
            InitializeComponent()

            Me.Title = "ControlBar"
            Me.Description = "ControlBar can provide a floating, translucent window for form." & vbCr & vbLf & "- ControlBar provides built-in actions, the map demo shows how to use the built-in actions." & vbCr & vbLf & "- Of course, customize the action is simple, keyboard demo shows how to costomize the actions."
        End Sub

        Private Sub button1_Click(sender As Object, e As EventArgs)
            Dim form As Form = New MapViewFormForControlBar()
            form.Show(Me)
        End Sub

        Private Sub button3_Click(sender As Object, e As EventArgs)
            Dim form As Form = New InputFormforControlBar()
            form.Show()
        End Sub
    End Class
End Namespace
